File: src/MinimalCS/Minimal/Actions/ToggleWindowVisibilityAction.cs
using System.Windows.Interactivity;

namespace ESRI.ArcGIS.SilverlightMapApp.Actions
{
    public class ToggleWindowVisibilityAction : TargetedTriggerAction<DraggableWindow>
    {
        protected override void Invoke(object parameter)
        {
            this.Target.IsOpen = !this.Target.IsOpen;
        }
    }
}

File: src/MinimalVB/MinimalVB/Actions/ToggleWindowVisibilityAction.vb
Imports System.Windows.Interactivity

Namespace ESRI.ArcGIS.SilverlightMapApp.Actions
		Public Class ToggleWindowVisibilityAction
			Inherits TargetedTriggerAction(Of DraggableWindow)
				Protected Overrides Sub Invoke(ByVal parameter As Object)
						Me.Target.IsOpen = Not Me.Target.IsOpen
				End Sub
		End Class
End Namespace
